lib/waistband/model.rb

```diff
@@ -0,0 +1,195 @@
+require 'digest/md5'
+require 'active_support/core_ext/object/blank'
+require 'active_model'
+
+module Waistband
+  class Model
+
+    extend ::ActiveModel::Callbacks
+
+    include ::ActiveModel::Validations
+    include ::ActiveModel::Validations::Callbacks
+
+
+    #################
+    # Class methods #
+    #################
+
+    class << self
+
+      attr_reader :type_name, :column_defaults
+
+      def index_name(name)
+        @index_name = name.to_s
+      end
+
+      def index_type(name)
+        @type_name = name.to_s
+      end
+
+      def index
```

Comment:
I like the idea.  The only concern would be multi type indexes.  I think
for those the mappings would need to be in the YML but  maybe they're not
exclusive.
On Feb 22, 2015 9:15 AM, "Mike Nelson" <notifications@github.com> wrote:

> In lib/waistband/model.rb
> <https://github.com/taskrabbit/waistband/pull/18#discussion_r25133983>:
>
> > +    # Class methods #
> > +    #################
> > +
> > +    class << self
> > +
> > +      attr_reader :type_name, :column_defaults
> > +
> > +      def index_name(name)
> > +        @index_name = name.to_s
> > +      end
> > +
> > +      def index_type(name)
> > +        @type_name = name.to_s
> > +      end
> > +
> > +      def index
>
> I was thinking more of a dsl that declares the schema, including index
> configs.
>
> index name: 'things' do
>   mapping 'field1', type: 'datetime'end
>
> Something along those lines. Clearly it would need to be thought through a
> little more than what I have.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskrabbit/waistband/pull/18/files#r25133983>.
>